### PR TITLE
[MASTG-TOOL-0015] Update drozer content and remove outdated references

### PR DESCRIPTION
**Pull request by cyberMilosz**
This is the first PR triggered by #2598. It focuses on rewriting the tool page itself, making sure it's accurate and removing outdated references.

Further work will need to be done to restore drozer instructions that were removed from the main MASTG body. As a result, #2598 should stay open for now.

### CODE REVIEW

File: tools/android/MASTG-TOOL-0015.md [hunk 1]
```diff
@@ -1,76 +1,158 @@
...
 
 Other resources where you might find useful information are:
 
-- [Official drozer User Guide](https://labs.withsecure.com/assets/BlogFiles/mwri-drozer-user-guide-2015-03-23.pdf "Drozer User Guide").
-- [drozer GitHub page](https://github.com/FSecureLABS/drozer "GitHub repo")
-- [drozer Wiki](https://github.com/FSecureLABS/drozer/wiki "drozer Wiki")
\ No newline at end of file
+- [Official drozer User Manual](https://labs.withsecure.com/tools/drozer "drozer User Manual")
+- [drozer GitHub page](https://github.com/WithSecureLabs/drozer "GitHub repo - drozer")
+- [drozer Agent GitHub page](https://github.com/WithSecureLabs/drozer "GitHub repo - drozer-agent")
```
Comment (sushi2k) [inline tools/android/MASTG-TOOL-0015.md:158]: This is the same URL as the link before. I would suggest to remove it.

File: tools/android/MASTG-TOOL-0015.md [hunk 2]
```diff
@@ -1,76 +1,158 @@
...
-$ dz> run scanner.provider.injection -a (package name)
 ```
+dz> help app.activity.start
+Attempting to run shell module
+usage: run app.activity.start [-h] [--action ACTION] [--category CATEGORY [CATEGORY ...]] [--component PACKAGE COMPONENT] [--data-uri DATA_URI] [--extra TYPE KEY VALUE] [--flags FLAGS [FLAGS ...]] [--mimetype MIMETYPE]
+```
+
+You can learn more about how intents are created by running `help intents`
+
+#### Further information
```
Comment (sushi2k) [inline tools/android/MASTG-TOOL-0015.md:148]: This link is already used in "### Example usage". Can you remove this short section and merge it with "### Example usage"?

File: tools/android/MASTG-TOOL-0015.md [hunk 3]
```diff
@@ -1,76 +1,158 @@
...
-$ dz> run app.broadcast.info -a (package name)
+#### Starting an Activity
+In order to launch an exported activity, use the following command:
+```
+run app.activity.start --component <package> <component name>
+```
 
-# Send a message to a Broadcast receiver
-$ dz> run app.broadcast.send --action (broadcast receiver name) -- extra (number of arguments)
+When calling `app.activity.start`, you can build a much more complex intent. As with all drozer modules, you can request more usage information:
```
Comment (sushi2k) [inline tools/android/MASTG-TOOL-0015.md:138]:
```suggestion
When calling `app.activity.start`, you can build a much more complex intent. As with all drozer modules, you can request more usage information by using the command `help`:
```

File: tools/android/MASTG-TOOL-0015.md [hunk 4]
```diff
@@ -1,76 +1,158 @@
...
 
-# Detect SQL injections in content providers
-$ dz> run scanner.provider.injection -a (package name)
 ```
+dz> help app.activity.start
+Attempting to run shell module
+usage: run app.activity.start [-h] [--action ACTION] [--category CATEGORY [CATEGORY ...]] [--component PACKAGE COMPONENT] [--data-uri DATA_URI] [--extra TYPE KEY VALUE] [--flags FLAGS [FLAGS ...]] [--mimetype MIMETYPE]
+```
+
+You can learn more about how intents are created by running `help intents`
```
Comment (sushi2k) [inline tools/android/MASTG-TOOL-0015.md:146]:
```suggestion
You can learn more about how intents are created by running `help intents`.
```

**Review by sushi2k — CHANGES_REQUESTED** [review]
Thank you for the pull request @cyberMilosz! This is a nice update for drozer. Please find a few comments and suggestions below. Let me know if any questions.

File: tools/android/MASTG-TOOL-0015.md (hunk 1 above)
Comment (cyberMilosz) [reply tools/android/MASTG-TOOL-0015.md:158]: Whoops, that second link is supposed to be https://github.com/WithSecureLabs/drozer-agent - will correct soon!

**Comment by cyberMilosz** [comment]
Thank you - will review ASAP and get back to you!

**Comment by sushi2k** [comment]
@cyberMilosz  You can just use the "Commit Suggestion" button if you agree to the proposed changes. Thanks!

File: tools/android/MASTG-TOOL-0015.md [hunk 5]
```diff
@@ -1,76 +1,158 @@
 ---
-title: Drozer
+title: drozer
 platform: android
-source: https://github.com/FSecureLABS/drozer
+source: https://github.com/WithSecureLabs/drozer
 ---
 
-[Drozer](https://github.com/FSecureLABS/drozer "Drozer on GitHub") is an Android security assessment framework that allows you to search for security vulnerabilities in apps and devices by assuming the role of a third-party app interacting with the other application's IPC endpoints and the underlying OS.
+[drozer](https://github.com/WithSecureLabs/drozer "drozer on GitHub") a security testing framework for Android. that allows you to search for security vulnerabilities in apps and devices by assuming the role of an app interacting with Android Runtime, other apps' IPC endpoints, and the underlying OS.
```
Comment (TheDauntless) [inline tools/android/MASTG-TOOL-0015.md:7]:
```suggestion
[drozer](https://github.com/WithSecureLabs/drozer "drozer on GitHub") is a security testing framework for Android that allows you to search for security vulnerabilities in apps and devices by assuming the role of an app interacting with the Android Runtime, other apps' IPC endpoints, and the underlying OS.
```

File: tools/android/MASTG-TOOL-0015.md [hunk 6]
```diff
@@ -1,76 +1,158 @@
...
+drozer helps to reduce the time taken for Android security assessments by automating tedious and time-consuming tasks. It allows testers and reverse-engineers to:
 
-## Installing Drozer
+- Discover and interact with the attack surface exposed by Android apps.
+- Execute dynamic Java-code on a device, to avoid the need to compile and install small test scripts.
 
-You can refer to [drozer GitHub page](https://github.com/FSecureLABS/drozer "Drozer on GitHub") (for Linux and Windows, for macOS please refer to this [blog post](https://fi5t.xyz/en/posts/drozer-on-mac/ "(not)Unique experience blog - Installing Drozer on macOS Catalina")) and the [drozer website](https://labs.withsecure.com/tools/drozer/ "Drozer Website") for prerequisites and installation instructions.
+drozer runs both in Android emulators and on real devices. It does not require USB debugging or other development features to be enabled; so you can perform assessments on devices in their production state to get better results.
```
[the remark this answers is not in the document]
Comment (TheDauntless) [reply tools/android/MASTG-TOOL-0015.md:14]: Agree with this change. You also don't need root for ADB and you don't even need ADB to install additional APKs.

File: tools/android/MASTG-TOOL-0015.md [hunk 7]
```diff
@@ -1,76 +1,158 @@
...
+      .isisandp..rotectorandroid..snemisis.
+      ,andprotectorandroidsnemisisandprotec.
+     .torandroidsnemesisandprotectorandroid.
+     .snemisisandprotectorandroidsnemesisan:
+     .dprotectorandroidsnemesisandprotector.
+
+drozer Console (v3.0.0)
+dz>
+```
+The prompt confirms the Android ID of the device you have connected to, along with the manufacturer, model and Android software version.
```
Comment (TheDauntless) [inline tools/android/MASTG-TOOL-0015.md:106]:
```suggestion
The prompt confirms the Android ID of the device you have connected to, along with the manufacturer, model and Android OS version.
```

**Review by TheDauntless — CHANGES_REQUESTED** [review]
A few additional suggested modifications. You can simply accept them directly in this PR and then we can merge everything :)

File: tools/android/MASTG-TOOL-0015.md (hunk 1 above)
Comment (cpholguera) [inline tools/android/MASTG-TOOL-0015.md:158]:
```suggestion
- [drozer Agent GitHub page](https://github.com/WithSecureLabs/drozer-agent "GitHub repo - drozer-agent")
```

**Review by sushi2k — APPROVED** [review]

File: tools/android/MASTG-TOOL-0015.md [hunk 8]
```diff
@@ -1,76 +1,76 @@
...
+#### Searching for applications on the device
 
-```bash
-dz> run app.activity.start --component <package> <component name>
+```
+run app.package.list -f <keyword>
 ```
 
-This previous command will start the activity, hopefully leaking some sensitive information. Drozer has modules for every type of IPC mechanism. Download [InsecureBankv2](0x08b-Reference-Apps.md#insecurebankv2) if you would like to try the modules with an intentionally vulnerable application that illustrates common problems related to IPC endpoints. Pay close attention to the modules in the scanner category as they are very helpful automatically detecting vulnerabilities even in system packages, specially if you are using a ROM provided by your cellphone company. Even [SQL injection vulnerabilities in system packages by Google](https://issuetracker.google.com/u/0/issues/36965126 "SQL injection in Android") have been identified in the past with drozer.
+This lists basic informations about any packages containing the word "<keyword>" in their bundle identifier. This includes package names, key directories used by the application, and any permissions used or defined by the application.
```
Comment (TheDauntless) [inline tools/android/MASTG-TOOL-0015.md:34]:
```suggestion
This lists basic information about any packages containing the word "<keyword>" in their package name. This includes the app name, key directories used by the application, and any permissions used or defined by the application.
```
Bundle name is only used on iOS IMO.

File: tools/android/MASTG-TOOL-0015.md [hunk 9]
```diff
@@ -1,76 +1,76 @@
...
+## Installing drozer and Setup
 
-Once the setup is completed you can start a session to an emulator or a device connected per USB by running `adb forward tcp:31415 tcp:31415` and `drozer console connect`. This is called direct mode and you can see the full instructions in the [User Guide in section "Starting a Session"](https://labs.withsecure.com/assets/BlogFiles/mwri-drozer-user-guide-2015-03-23.pdf "Starting a Session"). An alternative is to run Drozer in infrastructure mode, where, you are running a drozer server that can handle multiple consoles and agents, and routes sessions between them. You can find the details of how to setup drozer in this mode in the ["Infrastructure Mode"](https://labs.withsecure.com/assets/BlogFiles/mwri-drozer-user-guide-2015-03-23.pdf "Infrastructure Mode") section of the User Guide.
+Detailed instructions on how to install and setup the drozer console on your machine and the drozer agent on the Android phone can be found in the [drozer Github repo](https://github.com/WithSecureLabs/drozer "Installation instructions of drozer").
```
Comment (TheDauntless) [inline tools/android/MASTG-TOOL-0015.md:20]:
```suggestion
Detailed instructions on how to install and set up the drozer console on your machine and the drozer agent on the Android phone can be found in the [drozer Github repo](https://github.com/WithSecureLabs/drozer "Installation instructions of drozer").
```

**Review by TheDauntless — APPROVED** [review]

File: tools/android/MASTG-TOOL-0015.md (hunk 8 above)
Comment (cpholguera) [inline tools/android/MASTG-TOOL-0015.md:34]:
```suggestion
This lists basic information about any packages containing the word "<keyword>" in their bundle identifier. This includes package names, key directories used by the application, and any permissions used or defined by the application.
```

**Review by cpholguera — APPROVED** [review]

File: tools/android/MASTG-TOOL-0015.md [hunk 10]
```diff
@@ -1,76 +1,76 @@
...
 ```
 
-This previous command will start the activity, hopefully leaking some sensitive information. Drozer has modules for every type of IPC mechanism. Download [InsecureBankv2](0x08b-Reference-Apps.md#insecurebankv2) if you would like to try the modules with an intentionally vulnerable application that illustrates common problems related to IPC endpoints. Pay close attention to the modules in the scanner category as they are very helpful automatically detecting vulnerabilities even in system packages, specially if you are using a ROM provided by your cellphone company. Even [SQL injection vulnerabilities in system packages by Google](https://issuetracker.google.com/u/0/issues/36965126 "SQL injection in Android") have been identified in the past with drozer.
+This lists basic information about any packages containing the word "<keyword>" in their bundle identifier. This includes package names, key directories used by the application, and any permissions used or defined by the application.
 
-## Other Drozer commands
+#### Enumerating the attack surface of an app
 
-Here's a non-exhaustive list of commands you can use to start exploring on Android:
+```
```
Comment (cpholguera) [inline tools/android/MASTG-TOOL-0015.md:38]:
```suggestion
```sh
```

File: tools/android/MASTG-TOOL-0015.md [hunk 11]
```diff
@@ -1,76 +1,76 @@
...
-$ dz> run app.package.list
+This command inspects the target app's manifest and provides a report on any exported components of the application, and verifies whether the application is debuggable.
 
-# Find the package name of a specific app
-$ dz> run app.package.list -f (string to be searched)
+Once the attack surface has been identified, you can obtain more specific information about each component class. For example, to list  Activities, you can use the following command:
 
-# See basic information
-$ dz> run app.package.info -a (package name)
+```
```
Comment (cpholguera) [inline tools/android/MASTG-TOOL-0015.md:46]:
```suggestion
```sh
```

File: tools/android/MASTG-TOOL-0015.md [hunk 12]
```diff
@@ -1,76 +1,76 @@
...
-$ dz> run app.activity.info -a (package name)
+#### Starting an Activity
 
-# Launch the exported Activities
-$ dz> run app.activity.start --component (package name) (component name)
+In order to launch an exported activity, use the following command:
 
-# Identify the list of exported Broadcast receivers
-$ dz> run app.broadcast.info -a (package name)
+```
```
Comment (cpholguera) [inline tools/android/MASTG-TOOL-0015.md:56]:
```suggestion
```sh
```

File: tools/android/MASTG-TOOL-0015.md [hunk 13]
```diff
@@ -1,76 +1,76 @@
...
+run app.activity.start --component <package> <component name>
+```
 
-# Send a message to a Broadcast receiver
-$ dz> run app.broadcast.send --action (broadcast receiver name) -- extra (number of arguments)
+When calling `app.activity.start`, you can build a much more complex intent. As with all drozer modules, you can request more usage information by using the `help` command:
 
-# Detect SQL injections in content providers
-$ dz> run scanner.provider.injection -a (package name)
 ```
```
Comment (cpholguera) [inline tools/android/MASTG-TOOL-0015.md:62]:
```suggestion
```sh
```

**Pull request merged**